### PR TITLE
Feature/tmchips

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
Screenshot_1691999852
Screenshot_1691999859
Screenshot_1691999870

### CODE REVIEW

File: DemoApp/DemoApp/ViewModels/TMChipsPageViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace DemoApp.ViewModels
+{
+    internal partial class TMChipsPageViewModel : ObservableObject
+    {
+        private int count = 0;
+        public ObservableCollection<TMChips> ChipsCollection { get; } = new ObservableCollection<TMChips>();
+
+        public TMChipsPageViewModel()
+        {
+            ChipsCollection.Add(new TMChips() { Title = "Chip 1", Type = ChipType.Input, LeftIconSource = ImageConstants.AccountIcon, ChipsSize = ChipSize.Small });
```
Comment (rthanga1) [inline DemoApp/DemoApp/ViewModels/TMChipsPageViewModel.cs:18]: TMChip is a UI component. Shouldn't be here in the ViewModel. For this, create a DataModel class like a Person object or something. You can have the Title and the Image in that and you can hardcode the Type and the ChipsSize in the xaml as they are not changing

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+using Microsoft.Maui.Controls.Shapes;
+using System;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+using Trimble.Modus.Components.Constant;
+using Trimble.Modus.Components.Enums;
+using Trimble.Modus.Components.Helpers;
+
+public partial class TMChips : ContentView
+{
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:13]: Selected color is not right, It should be light blue opaque color. But here it is our Primary Trimble Blue color
selected-color

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    private EventHandler _clicked;
+
+    private TapGestureRecognizer tapGestureRecognizer = new TapGestureRecognizer();
+    #endregion
+    #region Bindable Properties
+    public static readonly BindableProperty LeftIconSourceProperty =
+        BindableProperty.Create(nameof(LeftIconSource), typeof(ImageSource), typeof(TMChips),propertyChanged:OnLeftIconPropertyChanged);
+
+    public static readonly BindableProperty TitleProperty =
+        BindableProperty.Create(nameof(Title), typeof(string), typeof(TMChips), propertyChanged: OnTextChanged);
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:25]: Renamed

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:13]: Its actually 30 percent of TrimbleBlue Color so added it as a new color for chip selected

File: DemoApp/DemoApp/ViewModels/TMChipsPageViewModel.cs (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/ViewModels/TMChipsPageViewModel.cs]: Added a Chips Item for the collection

File: DemoApp/DemoApp/DemoApp.csproj [hunk 4]
```diff
@@ -106,6 +106,9 @@
 	  <Compile Update="Views\TMButtonPage.xaml.cs">
 	    <DependentUpon>TMButtonPage.xaml</DependentUpon>
 	  </Compile>
+	  <Compile Update="Views\TMChipsPage.xaml.cs">
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/DemoApp.csproj:109]: Renamed

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+             WidthRequest="24"/>
+      <Label x:Name="label"
+             LineHeight="20"
+             FontAttributes="Bold"
+             VerticalOptions="Center"
+             HorizontalOptions="Center"/>
+      <ImageButton x:Name="righticon"
+                   BackgroundColor="Transparent"
+                   Clicked="OnRightIconTapped"
+             Source="{Static constants:ImageConstants.CancelCircle}"
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml:26]: Formatted

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+            }
+            else
+            {
+                tmChips.frame.Opacity = 1;
+            }
+            tmChips.UpdateTapGestureRecogniser();
+        }
+    }
+    private static void OnLeftIconPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+    {
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:236]: It has another property dependent (ChipType) hence its in code-behind.

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+        if (bindable != null && bindable is TMChips TMChips)
+        {
+            UpdateLabelOnSize(TMChips);
+        }
+    }
+    private static void OnTextChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable != null && bindable is TMChips TMChips)
+        {
+            TMChips.label.Text = (string)newValue;
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs]: The only property which can be binded is Text and sometimes in ios while binding text in xaml it doesn't appear in TMButton hrnce can we go with this approach?

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+          x:Name="frame"
+          HorizontalOptions="Start"
+          VerticalOptions="Center"
+          Stroke="Transparent"
+          >
+    <HorizontalStackLayout>
+      <Image x:Name="lefticon"
+             HeightRequest="24"
+             WidthRequest="24"/>
+      <Label x:Name="label"
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml:18]: Added OpenSans-SemiBold (weight:600)

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<ContentView xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
```
[the remark this answers is not in the document]
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml:2]: I fixed this

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs (hunk 2 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:13]: Looks like it is fixed now?

File: DemoApp/DemoApp/Models/ChipsItem.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Windows.Input;
+using Trimble.Modus.Components.Enums;
+
+namespace DemoApp.Models;
+public class ChipsItem
+{
+    public string Title { get; set; }
+    public ChipType Type { get; set; }
+    public ChipSize ChipSize { get; set; }
+    public ICommand ClickChipCommand { get; set; }
```
Comment (rthanga1) [inline DemoApp/DemoApp/Models/ChipsItem.cs:10]: ICommand should not be in the model class. It belong to the ViewModel

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Models/ChipsItem.cs (hunk 10 above)
Comment (rthanga1) [reply DemoApp/DemoApp/Models/ChipsItem.cs:10]: I have addressed this

**Review by rthanga1 — DISMISSED** [review]

**Review by rthanga1 — APPROVED** [review]

File: Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMChips/TMChips.xaml.cs:13]: Yes

**Pull request merged**